siege_utilities/geo/providers/nces_download.py

```
@@ -0,0 +1,388 @@
+        result = download_file(url, zip_path, timeout=self.timeout)
+        if not result:
+            raise NCESDownloadError(
+                f"Failed to download {data_type} for year {year} from {url}"
+            )
+
+        # Extract
+        extract_dir.mkdir(parents=True, exist_ok=True)
+        with zipfile.ZipFile(zip_path, "r") as zf:
+            zf.extractall(extract_dir)
+
+        log.info(f"Extracted {data_type} data to {extract_dir}")
```

Ruff (0.15.11)

[warning] Logging statement uses f-string

(G004)